cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -343,6 +344,17 @@ public long getApplicationCount() throws IOException {
     return getApplicationSpecificationTable().count(ranges);
   }
 
+  public ApplicationMeta getLatest(NamespaceId namespaceId, String appName) throws IOException {
+    // TODO : replace with scan by index. JIRA: CDAP-19656 
+    ApplicationMeta latest =  scanAppsWithRange(getNamespaceAndApplicationRange(namespaceId.getNamespace(), appName))
```

Comment:
this will scan everything in the range and put it all into memory. Instead of that, it should scan with limit 1 and just return the first row (if it exists) and close the iterator.

I don't think we need the scanAppsWithRange method either, it's only being called from this method. Since the getLatest method is already pretty short, it's actually more readable to do the scanning here rather than pulling it out into a separate method.